resources/tentative/complex-static-html/src/big-dom-generator/gen-css.js

```diff
@@ -1,238 +1,197 @@
+    const nextElement = getRandomElement(combinator, element, depth, nextDepth);
+
+    return buildSelectors(nextElement, nextDepth, combinator, selectorLength + 1, maxSelectorLength, isMatching) + selector + oldCombinator;
 };
 
-// Returns a non-matching selector for the element.
-// We kept the selector from matching by adding the `.just-span` class to the left-most selector or
-// by adding a classname from one of its children.
-// TODO: Is there a better way to ensure the selector is non-matching?
-const buildNonMatchingSelector = (depth, index, oldCombinator, selLen, badSelector) => {
-    // If we are in the top node, we are done.
-    // prettier-ignore
-    if (!depth)
-        return `.just-span${ oldCombinator}`;
-
-    // If we've reached the target length, pick a random classname from its children.
-    const getSelector = randomWeighted([getClassname, getType, () => "*"], [0.6, 0.3, 0.1]);
-    const selector = getSelector(depth, index);
-    if (selLen === badSelector) {
-        const wrongDepth = random.randRange(Math.min(depth + 1, 7), 8);
-        const wrongSelector = getClassname(wrongDepth, 0);
-        return selector + wrongSelector + oldCombinator;
-    }
+const ANGULAR_VIEW_DEPTH = 8;
+const ANGULAR_LI_DEPTH = 7;
+const NON_ANGULAR_VIEW_DEPTH = 6;
+const NON_ANGULAR_LI_DEPTH = 5;
 
-    // Otherwise, recurse.
-    const combinator = chooseCombinator(depth, index);
-    const nextDepth = getNextDepth(combinator, depth);
-    const nextIndex = getNextIndex(combinator, nextDepth, index);
-    return buildNonMatchingSelector(nextDepth, nextIndex, combinator, selLen + 1, badSelector) + selector + oldCombinator;
+const getInitialDepth = (element, isAngular) => {
+    if (isAngular)
+        return element.tagName === "DIV" ? ANGULAR_VIEW_DEPTH : ANGULAR_LI_DEPTH;
+
+    return element.tagName === "DIV" ? NON_ANGULAR_VIEW_DEPTH : NON_ANGULAR_LI_DEPTH;
+};
+
+// Take selectors create random color styles. Same color, different opacity.
+const generateCssRules = (selectors) => {
+    return selectors.map((selector, i) => {
+        random.shuffle(cssProperties, true);
+        return `${selector} {
+                    ${cssProperties[0]}: rgba(140,140,140,${i / 1000});
+                    ${cssProperties[1]}: rgba(140,140,140,${i / 1000});
+                }`;
+    });
 };
 
 const cssProperties = ["accent-color", "border-bottom-color", "border-color", "border-left-color", "border-right-color", "border-top-color", "column-rule-color", "outline-color", "text-decoration-color"];
 
-// Returns a random 200 matching selectors and 200 non-matching selectors targeted at the todoMVC items.
-export const genCss = () => {
+/**
+ * Returns a random 200 matching selectors and 200 non-matching selectors targeted at the todoMVC items.
+ * @param {string} isAngular whether to generate angular or react markup
+ * @returns {string} The css rules for the matching and non-matching selectors.
+ */
+export const genCss = (isAngular = false) => {
```

Comment:
I guess we can cross this bridge if/when it happens but if there are other special cases to handle then we'd want to pass some other type besides a boolean around.